File: app/src/main/java/com/app/customisedchartswithtooltip/utils/CustomCombinedChartRenderer.java
package com.app.customisedchartswithtooltip.utils;

import com.github.mikephil.charting.animation.ChartAnimator;
import com.github.mikephil.charting.charts.CombinedChart;
import com.github.mikephil.charting.renderer.BubbleChartRenderer;
import com.github.mikephil.charting.renderer.CandleStickChartRenderer;
import com.github.mikephil.charting.renderer.CombinedChartRenderer;
import com.github.mikephil.charting.renderer.LineChartRenderer;
import com.github.mikephil.charting.renderer.ScatterChartRenderer;
import com.github.mikephil.charting.utils.ViewPortHandler;

public class CustomCombinedChartRenderer extends CombinedChartRenderer {

    public CustomCombinedChartRenderer(CombinedChart chart, ChartAnimator animator, ViewPortHandler viewPortHandler) {
        super(chart, animator, viewPortHandler);
    }

    @Override
    public void createRenderers() {
        this.mRenderers.clear();
        CombinedChart chart = (CombinedChart)this.mChart.get();
        if (chart != null) {
            CombinedChart.DrawOrder[] orders = chart.getDrawOrder();
            CombinedChart.DrawOrder[] var3 = orders;
            int var4 = orders.length;

            for(int var5 = 0; var5 < var4; ++var5) {
                CombinedChart.DrawOrder order = var3[var5];
                switch(order) {
                    case BAR:
                        if (chart.getBarData() != null) {
                            this.mRenderers.add(new CustomBarChartRendererFromTop(chart, this.mAnimator, this.mViewPortHandler));
                        }
                        break;
                    case BUBBLE:
                        if (chart.getBubbleData() != null) {
                            this.mRenderers.add(new BubbleChartRenderer(chart, this.mAnimator, this.mViewPortHandler));
                        }
                        break;
                    case LINE:
                        if (chart.getLineData() != null) {
                            this.mRenderers.add(new LineChartRenderer(chart, this.mAnimator, this.mViewPortHandler));
                        }
                        break;
                    case CANDLE:
                        if (chart.getCandleData() != null) {
                            this.mRenderers.add(new CandleStickChartRenderer(chart, this.mAnimator, this.mViewPortHandler));
                        }
                        break;
                    case SCATTER:
                        if (chart.getScatterData() != null) {
                            this.mRenderers.add(new ScatterChartRenderer(chart, this.mAnimator, this.mViewPortHandler));
                        }
                }
            }    
        }
    }
}

File: app/src/main/java/com/app/customisedchartswithtooltip/utils/CustomCombinedChartRenderer.kt
package com.app.customisedchartswithtooltip.utils

import com.github.mikephil.charting.animation.ChartAnimator
import com.github.mikephil.charting.charts.CombinedChart
import com.github.mikephil.charting.charts.CombinedChart.DrawOrder
import com.github.mikephil.charting.renderer.*
import com.github.mikephil.charting.utils.ViewPortHandler

class CustomCombinedChartRenderer(
    chart: CombinedChart?,
    animator: ChartAnimator?,
    viewPortHandler: ViewPortHandler?
) : CombinedChartRenderer(chart, animator, viewPortHandler) {
    override fun createRenderers() {
        mRenderers.clear()
        val chart = mChart.get() as CombinedChart?
        if (chart != null) {
            val orders = chart.drawOrder
            val var4 = orders.size
            for (var5 in 0 until var4) {
                val order = orders[var5]
                when (order) {
                    DrawOrder.BAR -> if (chart.barData != null) {
                        mRenderers.add(
                            CustomBarChartRendererFromTop(
                                chart,
                                mAnimator,
                                mViewPortHandler
                            )
                        )
                    }
                    DrawOrder.BUBBLE -> if (chart.bubbleData != null) {
                        mRenderers.add(
                            BubbleChartRenderer(
                                chart,
                                mAnimator,
                                mViewPortHandler
                            )
                        )
                    }
                    DrawOrder.LINE -> if (chart.lineData != null) {
                        mRenderers.add(
                            LineChartRenderer(
                                chart,
                                mAnimator,
                                mViewPortHandler
                            )
                        )
                    }
                    DrawOrder.CANDLE -> if (chart.candleData != null) {
                        mRenderers.add(
                            CandleStickChartRenderer(
                                chart,
                                mAnimator,
                                mViewPortHandler
                            )
                        )
                    }
                    DrawOrder.SCATTER -> if (chart.scatterData != null) {
                        mRenderers.add(
                            ScatterChartRenderer(
                                chart,
                                mAnimator,
                                mViewPortHandler
                            )
                        )
                    }
                }
            }
        }
    }
}